workbench-core/example/infrastructure/integration-tests/tests/authorization/routeProtection.test.ts

```diff
@@ -0,0 +1,150 @@
+import { IdentityPermission } from '@aws/workbench-core-authorization';
+import { fc, testProp } from 'jest-fast-check';
+import ClientSession from '../../support/clientSession';
+import Setup from '../../support/setup';
+import HttpError from '../../support/utils/HttpError';
+
+describe('Static authorization integration tests', () => {
+  const setup: Setup = new Setup();
+  let adminSession: ClientSession;
+  let newAdminSession: ClientSession;
+
+  beforeEach(() => {
+    expect.hasAssertions();
+  });
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+    const { data } = await adminSession.resources.groups.create();
+    const groupId = data.groupId;
+    const auditPermissionsRequired: IdentityPermission[] = [
```

Comment:
You should have a test with conflicting ALLOW/DENY permissions to ensure DENY wins.